lib/api/apiUtils/object/createAndStoreObject.js

```diff
@@ -162,13 +163,14 @@ function createAndStoreObject(bucketName, bucketMD, objectKey, objMD, authInfo,
                 });
         },
         function storeMDAndDeleteData(options, infoArr, next) {
+            const noDelete = objMD ? objMD.noDelete : null;
             metadataStoreParams.versionId = options.versionId;
             metadataStoreParams.versioning = options.versioning;
             metadataStoreParams.isNull = options.isNull;
             metadataStoreParams.nullVersionId = options.nullVersionId;
             return _storeInMDandDeleteData(bucketName, infoArr,
                 cipherBundle, metadataStoreParams,
-                options.dataToDelete, requestLogger, next);
+                options.dataToDelete, requestLogger, noDelete, next);
```

Comment:
Please add tests for putting objects with 0 bytes on aws to make sure the metadata is there and can get both directly from aws and through scality s3.